app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -47,9 +47,9 @@ def fetch_middleware_servers
 
       def fetch_domains_with_servers
         @data[:middleware_domains] = []
-        @data[:middleware_server_groups] = []
         @ems.feeds.each do |feed|
           @ems.domains(feed).each do |domain|
+            @data[:middleware_server_groups] = []
```

Comment:
At this point, it looks like the `@data[:middleware_server_groups]` hash entry is really just a transient variable used in this loop, since it's getting reset on each iteration.

Maybe it's better to just have `fetch_server_groups(feed)` return the `middleware_server_groups` so they can be assigned to [`parsed_domain[:middleware_server_groups]`](https://github.com/ManageIQ/manageiq/pull/13321#discussion_r95403298) a couple lines later.